ios-target/View/Home/TopicsViewController.swift

```diff
@@ -0,0 +1,83 @@
+//
+//  TopicsViewController.swift
+//  ios-target
+//
+//  Created by sol manini on 7/1/19.
+//  Copyright © 2019 TopTier labs. All rights reserved.
+//
+
+import UIKit
+
+class TopicsViewController: ModalViewController {
+  
+  @IBOutlet weak var topicsCollectionHeightContraint: NSLayoutConstraint!
+  @IBOutlet weak var topicsCollectionView: UICollectionView!
+  
+  var viewModel = TopicsViewModel()
+  
+  weak var delegate: TopicViewModelDelegate?
+  
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    viewModel.delegate = self
+    setupView()
+  }
+  
+  func setupView() {
+    topicsCollectionHeightContraint.constant = CGFloat(self.viewModel.collectionHeight)
+    topicsCollectionView.register(UINib(nibName: TopicsCollectionViewCell.reuseIdentifier,
+                                        bundle: nil), forCellWithReuseIdentifier: TopicsCollectionViewCell.reuseIdentifier)
+    topicsCollectionView.dataSource = self
+    topicsCollectionView.delegate = self
+  }
+}
+
+extension TopicsViewController: TopicViewModelDelegate {
+  func topicSelected(topic: Topic) {
+    delegate?.topicSelected(topic: topic)
+  }
+  
+  func didUpdateState() {
+    if viewModel.state == .loading {
```

Comment:
This snippet is on every viewController, maybe you can do an `UIApplication` extension like `toggleNetworkActivity(_ active: Bool)` to extract this